packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2465,27 +2526,58 @@ export function writeCompletedBoundaryInstruction(
     );
   }
 
+  // Write function arguments, which are string and array literals
   const formattedContentID = stringToChunk(contentSegmentID.toString(16));
   writeChunk(destination, boundaryID);
-  writeChunk(destination, completeBoundaryScript2);
+  if (scriptFormat) {
+    writeChunk(destination, completeBoundaryScript2);
+  } else {
+    writeChunk(destination, completeBoundaryData2);
+  }
   writeChunk(destination, responseState.segmentPrefix);
   writeChunk(destination, formattedContentID);
   if (enableFloat && hasStyleDependencies) {
-    writeChunk(destination, completeBoundaryScript2a);
-    writeStyleResourceDependencies(destination, boundaryResources);
+    // Script and data writers must format this differently:
+    //  - script writer emits an array literal ["1", "2"]
+    //  - data writer emits a string literal ["1", "2"])
+    if (scriptFormat) {
+      writeChunk(destination, completeBoundaryScript3a);
+      // boundaryResources encodes an array literal
+      writeStyleResourceDependenciesInJS(destination, boundaryResources);
+    } else {
+      writeChunk(destination, completeBoundaryData3aStart);
+      writeStyleResourceDependenciesInAttr(destination, boundaryResources);
+      writeChunk(destination, completeBoundaryData3aEnd);
+    }
+  } else {
+    if (scriptFormat) {
+      writeChunk(destination, completeBoundaryScript3b);
+    } else {
+      writeChunk(destination, completeBoundaryData3b);
+    }
+  }
+  if (scriptFormat) {
```

Comment:
I think you can host the scriptFormat condition above the float+styleDependency one so you only need to check it once

Reply:
> I think you can hoist the scriptFormat condition above the float+styleDependency one so you only need to check it once

I was hoping to colocate the writes (of different formats) so that these + future changes can be local, and maybe achieve some code sharing. 

If this doesn't make sense, happy to completely refactor each `writeXInstruction` function into `writeXInstructionScript` and `writeXInstructionData` (similar to what we did for `writeStyleResourceAttribute`). 
(If this is the best pattern, can it wait to be addressed in a followup PR?)

Reply:
I think the general idea is code sharing is less important than branch elimination (caveats abound of course). This is maybe less true sometimes on the client where bundle size is more sensitive and it is also true we don't want excessive code size on the server but I think in the long run separate functions may be the way to go. Yeah separate PR followup is fine